client/asset/interface.go

```diff
@@ -897,10 +897,40 @@ type Balance struct {
 	// Locked is the total amount locked in the wallet which includes but
 	// is not limited to funds locked for swap but not actually swapped yet.
 	Locked uint64 `json:"locked"`
-	// Other is a place to list custom balance categories.
-	Other map[string]uint64 `json:"other"`
+	// Other is a place to list custom balance categories. It is recommended for
+	// custom balance added here to have a translation and tooltip info in
+	// client/webserver/site/src/js/wallet.js#customWalletBalanceCategory
+	Other map[BalanceCategory]CustomBalance `json:"other"`
 }
 
+// CustomBalance is a balance category used to track funds for a particular
+// purpose or for a special kind of balance (e.g Zcash Shielded wallet
+// balances).
+type CustomBalance struct {
+	// Amount is the balance in the wallet for this custom category. It is
+	// subtracted from Balance.Available above.
+	Amount uint64 `json:"amt"`
+	// Locked is a flag to indicate that this Amount is not included in
+	// Balance.Available and is included in Balance.Locked.
+	Locked bool `json:"locked"`
+}
+
+// BalanceCategory is a string identifier for a custom balance category.
+type BalanceCategory string
+
+// Balance categories for custom balances These values are used as a map key for
+// custom balances and may be recognized in the frontend to support translation.
+// It is recommended for custom balance categories listed here to have a
+// translation and tooltip info in
+// client/webserver/site/src/js/wallet.js#customWalletBalanceCategory. If any of
+// these balance categories should change, the customWalletBalanceCategory
+// function in the wallet.js file above should be updated with the new value.
+const (
+	BalanceCategoryBondReserves    = "Bond Reserves"
+	BalanceCategoryReservesDeficit = "Reserves Deficit"
```

Comment:
IMO these two should be fields of `asset.Balance` because these can be non-zero for (eventually) any asset, unlike shielded which is very specific to Zcash.

Reply:
As per my other comment, the idea is we don't add these to the map when they are zero because they are very often zero and it's going to be a waste of space on frontend.  So if we go this route and put them in asset.Balance, frontend just has to know to omit these from the table display if they are zero (IMO), because we don't want them displayed for every asset.

Reply:
> frontend just has to know to omit these from the table display if they are zero

This is what I figured we would do. Makes more sense as fields of the struct, since the front end doesn't need to guess whether they need to be displayed or not. The `Other` field is for custom balances categories that are unique to a particular asset, e.g. shielded, staked. I would not expect the front end to try to reason about whether to show zero balances from `Other`. If the asset chooses to include the entry, it should be displayed.

Reply:
Makes sense to me.

Reply:
Updated.